src/dtemplate.d

```diff
@@ -4059,27 +4062,46 @@ extern (C++) MATCH deduceType(RootObject o, Scope* sc, Type tparam, TemplatePara
             {
                 result = m;
             }
+            else if (e.type.toBasetype().isintegral() ||
```

Comment:
OK, done.

In D, only three types (`Tclass`, `Taarray`, and `Tstruct` with `hasPointers() == true`) cannot support the top-qualifier conversion. If `e.type` is not one of them, we IFTI can deduce corresponding type parameter.